### DISCUSSION TITLE
Copying Dart runtime in Dockerfile results in "cannot copy to non-directory" error

### DISCUSSION THREAD

**Discussion by adityapandeyz** [serverpod/serverpod#1712] [Q&A] [answered]
Issue Description:

When building the Docker image using the provided Dockerfile, an error occurs during the COPY operation from the build stage to the final stage. The error message indicates that there is an attempt to copy to a non-directory.

Steps to Reproduce:

1. Clone the repository.
2. Navigate to the directory containing the Dockerfile.
3. Run the docker build -t serverpod-app . command.

Expected Behavior:

The Docker image should be built successfully without any errors.

Actual Behavior:

The build process fails with the following error message:

`ERROR: failed to solve: cannot copy to non-directory: /var/lib/docker/overlay2/qeimp06mtt16u6pl5hph5zvdw/merged/lib64`

Dockerfile Section Causing the Issue:

`COPY --from=build /runtime/ /app/`

![image](https://github.com/serverpod/serverpod/assets/40023090/23da50f3-3af2-4b15-b17b-fa80e03934f9)

Complete dockerfile:

```
FROM dart:stable AS build

WORKDIR /app
COPY . .

RUN dart pub get
RUN dart compile exe bin/main.dart -o bin/main

FROM busybox

ENV runmode=development
ENV serverid=default
ENV logging=normal
ENV role=monolith

# COPY --from=build /runtime/ /

COPY --from=build /runtime/ /app/
COPY --from=build /app/bin/main /app/bin/main
COPY --from=build /app/config/ config/
COPY --from=build /app/generated/ generated/
COPY --from=build /app/web/ web/

EXPOSE 8080
EXPOSE 8081
EXPOSE 8082

CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role
```

**Comment by ewann** [comment, marked as answer]
there are now a couple of possible workarounds over in https://github.com/serverpod/serverpod/issues/1729
